src/lib.rs

```diff
@@ -21,14 +21,41 @@ pub(crate) fn app_dir(app_file: impl AsRef<Path>) -> Result<PathBuf> {
     Ok(path_buf)
 }
 
-pub(crate) fn write_failed_msg(bindle_id: &bindle::Id, dest_dir: &Path) -> String {
+pub(crate) fn push_all_failed_msg(path: &Path, server_url: &str) -> String {
     format!(
-        "Failed to write bindle '{}' to {}",
-        bindle_id,
-        dest_dir.display()
+        "Failed to push bindle from '{}' to the server at '{}'",
+        path.display(),
+        server_url
     )
 }
 
+pub(crate) fn missing_build_artifact_msg(path: impl AsRef<Path>) -> String {
+    format!(
+        "Component build artifact is missing: {}\n\nPlease try to run `spin build` first",
+        path.as_ref().display()
+    )
+}
+
+pub(crate) async fn prepare_bindle(
```

Comment:
Having two `prepare_bindle` functions is confusing to me - which do I use where?  I'd suggest having just one, the one in `spin_publish`, and here having a `to_printable_error` function (badly named, sorry).  Callers of this would then look like `spin_publish::prepare_bindle(...).await.map_err(to_printable_error)` which, although more verbose, makes the additional behaviour easier to see at the call site.

Reply:
done